core/arch/arm/include/sm/optee_smc.h

```diff
@@ -93,6 +93,12 @@
 #define OPTEE_SMC_SHM_CACHED		1
 
 /*
+ * Privileged call flag. Used by supplicant. Asks OP-TEE to use reserved
+ * thread if all regular threads are used.
+ */
+#define OPTEE_SMC_FLAGS_PRIVILEGED	BIT(0)
```

Comment:
I suppose it's possible with careful programming to avoid needing more than one reserved thread, but that means constantly being aware of this restriction and thus accumulating technical debt. Since this will affect the ABI it's important to get it right from the start.

But regardless wasting 8k for this is too much, especially since there's other options.